### ISSUE TITLE
Previous diagnostics are not removed

### ISSUE THREAD

**Issue by mattacosta** [Microsoft/vscode#56047]
Issue Type: <b>Bug</b>

After typing the colon in a new property, old diagnostics start to pile up in the problems view. Sometimes they don't even appear under the collapsible file name.

![diagnostics](https://user-images.githubusercontent.com/1007463/43870237-65330b24-9b2b-11e8-9199-28b42d167241.png)

In this case, when I opened up the developer tools panel I also noticed that:
- The list was at least twice as long, but those additional entries could not be scrolled to.
- Scrolling up and down somehow managed to add more diagnostics to the list?

On the one hand, this might be a Typescript issue since I've only noticed it with property declarations (FWIW I didn't find any related issues there either), but on the other hand there's some really weird UI behavior going on, so I'll let you guys figure it out.

Sample code:
```ts
'use strict';

export class A {

  protected b
  
  constructor() {
    // nothing
  }

}
```

Steps to reproduce:
1. Type in a `:` after the property name.
2. As you finish the declaration to remove the diagnostic and/or create new syntax errors, you can see the old diagnostics show up.

VS Code version: Code - Insiders 1.26.0-insider (c827480be25bd63065e7ee2e106c24a707c08004, 2018-08-08T18:51:13.457Z)
OS version: Windows_NT x64 10.0.17134

Extensions: none

**Comment by mattmccutchen** [comment]
Indeed, I noticed soon after upgrading today that something is very broken with the Problems view.  I'm going back to VS Code 1.25.1 for now.

**Comment by mattacosta** [comment]
This is no longer reproducible and was probably fixed by 26bd850.

If so, then #56371 is actually a duplicate of this issue. I'm going to close this one though, since the other issue has all the interesting bits. Feel free to reopen if necessary.
